read_montage and read_dig_montage now use the same nose fiducial name…

noreun opened the pull request:
Following [issue 3699](https://github.com/mne-tools/mne-python/issues/3699), the fiducials names should be standardized.

Eric89GXL commented:
Is there some test you could update or change that would fail on `master` but pass on this PR?

noreun commented:
Ok, I changed test_montage.py.

Before it only tested Transform=True for kind == '.htps', where fiducials should be ('2', '1', '3')

Now it tests a different type, where fiducials are assumed to be ('nasion', 'lpa', 'rpa')

Eric89GXL commented:
You have a style problem, a `# noqa:E501` should fix it.

Otherwise this LGTM, but @teonbrooks @jona-sassenhagen do you have time to look?

noreun commented:
I have a PEP8 plugin but apparently it was using 120 columns, sorry about that.

agramfort commented:
travis is still not happy

noreun commented:
Sorry I'm new to travis, and he seems to be quite a demanding individual. 

It shows this error:

```
======================================================================
FAIL: Test importing EEGLAB .set files
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/numpy/testing/decorators.py", line 147, in skipper_func
    return f(*args, **kwargs)
  File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/numpy/testing/decorators.py", line 147, in skipper_func
    return f(*args, **kwargs)
  File "/home/travis/build/mne-tools/mne-python/mne/io/eeglab/tests/test_eeglab.py", line 42, in test_io_set
    assert_equal(len(w1), 20)
AssertionError: 21 != 20
```

But when I run test_eeglab.py localy in my machine it passes:

https://drive.google.com/open?id=0B0p7WZ2wlTFSTFhIaUFoWnJhS2M

And indeed there are 20 warnings in w1:

https://drive.google.com/open?id=0B0p7WZ2wlTFSS0hWVTBOMWVmOW8

Do I need to do something like update the dataset I'm using locally in order to reproduce the error so I can debug it? I really don't know why test_eeglab.py is breaking since I didn't find any direct use of the function I changed.

More generally, is this the correct pipeline? Change, commit, push and look at travis output?
I tried to find a way of running travis locally but it doesn't seem trivial.

http://stackoverflow.com/questions/21053657/how-to-run-travis-ci-locally

Thanks!

jona-sassenhagen commented:
Argh. @jasmainak do you maybe have time to look?

The "catch 20 warnings" way probably wasn't the best way of testing the EEGLAB readers :(

jasmainak commented:
I think you should run the entire test suite locally to reproduce. It is probably some uncaught warning leaking into the eeglab test files ...

teonbrooks commented:
didn't realize that my comments were sent until i approved it. it's been a while since i've been on github. everything has changed! 😬

noreun commented on mne/channels/tests/test_montage.py:
```diff
@@ -103,7 +103,7 @@ def test_montage():
             assert_array_less(0.01, distance_from_centroid)
 
-    # test transform
+    # test transform with hpts [ fiducials == ('2', '1', '3') ]
```
pg 228 you mean? ;) 
http://www.martinos.org/meg/manuals/MNE-manual-2.7.pdf

teonbrooks replied:
hehe, I meant the pg 231 in the pdf using absolute numbering versus the labeled page number

agramfort commented:
@noreun you messed up this branch with master. Now the diff includes work from other PR

to avoid this don't do:

- git pull   # when you're not in master branch
- git merge master  # ie merge master in a feature branch.

if you need to get commits from master you need to rebase

- git checkout master
- git pull upstream master
- git checkout feature_branch
- git rebase master

please clean up.

thanks

noreun commented:
@agramfort That is really weird. I did not pull, actually I did rebase because I'm closely following this page

http://mne-tools.github.io/stable/contributing.html

what did happen is that for some reason a conflict appeared in test_montage.py during the rebase, and I had to clean it up. Apparently it did not work. I tried again with a git reset following this

http://mne-tools.github.io/stable/contributing.html#recovering-from-mess-up

And the commits looks better, although apparently  "continuous-integration/appveyor/pr" failed.. :/
I'm waiting for travis to finish.

Anyway, I just changed "nz" to "nasion" in read_montage so it behaves the same as read_dig_montage. Now with DigMontage dealing with electrodes in issue #3730, perhaps we don't need this PR anymore. Although it might be good to have the names of fiducials standardized...

agramfort commented:
@noreun  you need to rebase

noreun commented:
@agramfort rebased. Had a small conflict easy to solve.

@teonbrooks I added the references and fixed the typo.

Again, this is kind of cosmetics, or more like standardization.
But I think it is important that the default nasion has the same name everywhere in the system (avoids small hacks such as 

labels[labels.index('nz')] = 'nasion'

and back, which can be confusing.

Eric89GXL commented:
LGTM, +1 for merge

@teonbrooks can you look to see if you're happy now?

teonbrooks commented on mne/channels/montage.py:
```diff
@@ -298,7 +298,7 @@ def read_montage(kind, ch_names=None, path=None, unit='m', transform=False):
             fids = ('2', '1', '3')  # Alternate cardinal point names
         else:
-            fids = ('nz', 'lpa', 'rpa')
+            fids = ('nasion', 'lpa', 'rpa')
```
I think that this will be fine. the current state is actually a deviation from the [docstring](https://github.com/mne-tools/mne-python/blob/master/mne/channels/montage.py#L158) on the name of the fiducials in the montage file for the transformation to work

teonbrooks commented on mne/channels/tests/test_montage.py:
```diff
@@ -122,7 +123,24 @@ def test_montage():
     with open(fname, 'w') as fid:
         fid.write(input_str)
-    montage = read_montage(op.join(tempdir, 'test_fid.hpts'), transform=True)
+    montage = read_montage(op.join(tempdir, kind), transform=True)
```
could you add the test for the `.hpts`? the changes below fixes the test for `.sfp` but doesn't check the former.

agramfort commented:
what's our status here?

Eric89GXL commented:
@noreun any time to come back to this? Looks like it just needs a little cleanup

agramfort commented:
@teonbrooks can you maybe take over?

noreun commented:
Hello all, sorry I disappeared. Switched projects, then vacations...

I tried to address the comments and run the test for both hpts and sfp. Hope this helps. I had to rebase so now I'm waiting for he tests.

Just a recap, this PR is just about standardization: Today the Montage object expects the nose fiducial to be called 'nz', while the DigMontage objects expect it to be called 'nasion'. So the same file containing both electrodes positions and fiducials (like mine), wont work!

If this is not a problem, we can drop this PR. I would just need to find/replace it every time I call set_montage for one and for the other.

agramfort commented:
thx @noreun. Can you just add an entry in the what's new?

noreun commented:
Added to API since mostt changes in 'ChangeLog' are 'Add' something. Not sure if it is correct though.

agramfort commented:
thx @noreun
